ISPN-12444 Stream K/V endpoint (#8821)
infinispan/infinispan: documentation/src/main/asciidoc/topics/ref_rest_caches.adoc

|`limit`
|OPTIONAL
|Specifies the maximum number of keys to be streamed. A negative value will retrieve the keys. The default value is `-1`.
|`batch`
|OPTIONAL
|Specifies the internal batch size when retrieving the keys. The default value is `1000`.
|===

[id='rest_v2_get_entries']
= Retrieving All Entries from Caches
Invoke `GET` requests to retrieve all the entries in a cache in JSON format.

[source,options="nowrap",subs=attributes+]
----
GET /rest/v2/caches/{cacheName}?action=entries
----

.Request Parameters

|===
|Parameter |Required or Optional |Value

|`metadata`
|OPTIONAL
|Specifies if each entry should contain the metadata. The default value is `false`.
|`limit`
|OPTIONAL
|Specifies the maximum number of keys to be streamed. A negative value will retrieve the keys. The default value is `-1`.
|`batch`
|OPTIONAL
|Specifies the internal batch size when retrieving the keys. The default value is `1000`.
|===

{brandname} provides a JSON response such as the following:

[source,json,options="nowrap",subs=attributes+]
----
include::json_examples/rest_caches_all_entries.json[]
----

* `key` The key of the entry.
* `value` The value of the entry.

> * `lifespan` Value in milliseconds. -1 indicated not lifespan time. The value will not be present if metadata is parameter is `false`.

Reviewer: ```suggestion
* `lifespan` The entry lifespan in milliseconds. A value of `-1` means the stream does not include entry metadata.
```

> * `lifespan` Entry lifespan, in milliseconds, or `-1` if not specified. No value is returned unless you set metadata="true".

Reviewer: ```suggestion
* `lifespan` Lifespan, in milliseconds, or `Forever` if entries never expire. Returns a value of `-1` unless you set metadata="true".
```

> * `maxIdle` Value in milliseconds. -1 indicated not maxIdle time. The value will not be present if metadata is parameter is `false`.

Reviewer: ```suggestion
* `maxIdle` The entry maximum idle time in milliseconds. A value of `-1` means the stream does not include entry metadata.
```

> * `maxIdle` Maximum idle time, in milliseconds, for the entry or `-1` if not specified. No value is returned unless you set metadata="true".

Reviewer: ```suggestion
* `maxIdle` Maximum idle time, in milliseconds, or `Forever` if entries never expire. Returns a value of `-1` unless you set metadata="true".
```

> * `created` Value in milliseconds. -1 indicated not created time. The value will not be present if metadata is parameter is `false`.

Reviewer: ```suggestion
* `created` The entry creation time. A value of `-1` means the stream does not include entry metadata.
```

> * `created` Time the entry was created. Returns a value of `-1` if there is no creation time. No value is returned unless you set metadata="true".

Reviewer: ```suggestion
* `created` Time the entry was created. Returns a value of `-1` unless you set metadata="true".
```

> * `lastUsed` Value in milliseconds. -1 indicated not last used time. The value will not be present if metadata is parameter is `false`.

Reviewer: ```suggestion
* `lastUsed` The last time an operation was performed on the entry. A value of `-1` means the stream does not include entry metadata.
```

> * `lastUsed` The last time an operation was performed on the entry. Returns a value of `-1` if there is no last used time. No value is returned unless you set metadata="true".

Reviewer: ```suggestion
* `lastUsed` Last time an operation was performed on the entry. Returns a value of `-1` unless you set metadata="true".
```

> * `expireTime` Value in milliseconds. -1 indicated not expiration time. The value will not be present if metadata is parameter is `false`.

Reviewer 2: Same wordind needs to be changed in those descriptions. E.g.:

```suggestion
* `expireTime` Value in milliseconds, or -1 if the entry does not expire. The value will not be present if metadata is parameter is `false`.
```

Reviewer: What exactly is the expire time for an entry?

> * `expireTime` Value in milliseconds. -1 indicated not expiration time. The value will not be present if metadata is parameter is `false`.

Reviewer: ```suggestion
* `expireTime` The entry expiration time in milliseconds. A value of `-1` means the stream does not include entry metadata.
```

> * `expireTime` The entry expiration time in milliseconds. A value of `-1` means the stream does not include entry metadata.

Reviewer: ```suggestion
* `expireTime` Time when the entry expires or `Never` for immortal entries. Returns a value of `-1` unless you set metadata="true".
```
